new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CleanupUtil.java

```diff
@@ -0,0 +1,524 @@
+public class CleanupUtil {
+
+  /**
+   * Cleanup all artifacts in the Kubernetes cluster.
+   *
+   * <p>Tries to gracefully delete the WebLogic domains and WebLogic Operator in the namespaces, if found.
+   * Then deletes everything in the namespaces.
+   *
+   * <p>Waits for the deletion task to be completed until up to 3 minutes.
+   *
+   * @param namespaces list of namespaces
+   */
+  public static void cleanup(List<String> namespaces) {
+    try {
+      // If namespace list is empty or null return
+      if (namespaces == null || namespaces.isEmpty()) {
+        return;
+      }
+      // iterate through the namespaces and delete domain as a
+      // first entity if its not operator namespace
+      for (var namespace : namespaces) {
+        if (!isOperatorNamespace(namespace)) {
+          deleteDomains(namespace);
+        }
+      }
+      // iterate through the namespaces and delete operator if
+      // its operator namespace.
+      for (var namespace : namespaces) {
+        if (isOperatorNamespace(namespace)) {
+          uninstallOperator(namespace);
+        }
+      }
+
+      // Delete artifacts in namespace used by the test class
+      for (var namespace : namespaces) {
+        deleteNamespacedArtifacts(namespace);
+      }
+
+      Thread.sleep(30 * 1000); // Using Thread.sleep for a one time 30 sec sleep.
+      // If pollDelay is set to 30 seconds below, its going to sleep 30 seconds for every namespace.
+
+      // wait for the artifacts to be deleted, waiting for a maximum of 3 minutes
+      ConditionFactory withStandardRetryPolicy = with().pollDelay(0, SECONDS)
+          .and().with().pollInterval(10, SECONDS)
+          .atMost(3, MINUTES).await();
+
+      for (var namespace : namespaces) {
+        logger.info("Check for artifacts in namespace {0}", namespace);
+        withStandardRetryPolicy
+            .conditionEvaluationListener(
+                condition -> logger.info("Waiting for artifacts to be deleted in namespace {0}, "
+                    + "(elapsed time {1} , remaining time {2}",
+                    namespace,
+                    condition.getElapsedTimeInMS(),
+                    condition.getRemainingTimeInMS()))
+            .until(artifactsDoesntExist(namespace));
+      }
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
+      logger.warning("Cleanup failed");
+    }
+  }
+
+  /**
+   * Delete domains in the given namespace, if exists.
+   *
+   * @param namespace name
+   */
+  private static void deleteDomains(String namespace) {
+    try {
+      for (var item : Kubernetes.listDomains(namespace).getItems()) {
+        String domainUid = item.getMetadata().getName();
+        Kubernetes.deleteDomainCustomResource(domainUid, namespace);
+      }
+    } catch (Exception ex) {
+      logger.severe(ex.getMessage());
+      logger.severe("Failed to delete domain or the {0} is not a domain namespace", namespace);
+    }
+  }
+
+  /**
+   * Uninstalls the operator.
```

Comment:
uninstall the operator

Reply:
feels the the release name should be a param, or you should call this uninstallWebLogicOperators (plural)

Reply:
changed the method name

Reply:
> 
> 
> uninstall the operator

fixed it